### PR TITLE
Only auto-enable internals when necessary

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR changes the internals `StartingGearEquippedEvent` logic so that it only enables internals when required, and only if the linked tank actually allows the entity to survive. Should hopefully stop the random test failures.

The most dubious code is in `RespiratorSystem.GetSaturation()`, as that bodges some of the reagent effect condition checks, which generally require a full `ReagentEffectArgs` struct.

### CODE REVIEW

**Comment by Flareguy** [comment]
Does this account for toxic gasses in the air? I haven't read the code but I could imagine this causing issues if, say, a Vox spawned but didn't turn on internals due to the fact that there's Nitrogen in the airmix, ignoring the oxygen.

**Comment by ElectroJr** [comment]
> Does this account for toxic gasses in the air?

Not yet, but I can add that

**Comment by ElectroJr** [comment]
Should now check for toxic gasses as well, though GetSaturation is still pretty janky

File: Content.Server/Body/Systems/RespiratorSystem.cs [hunk 1]
```diff
@@ -164,6 +172,112 @@ public void Exhale(EntityUid uid, BodyComponent? body = null)
...
+            if (reagent.Metabolisms == null)
+                continue;
+
+            if (!reagent.Metabolisms.TryGetValue(GasId, out var entry))
+                continue;
+
+            foreach (var effect in entry.Effects)
+            {
+                if (effect is HealthChange health)
+                    toxic |= CanMetabolize(health) && health.Damage.Any();
```
Comment (EmoGarbage404) [inline Content.Server/Body/Systems/RespiratorSystem.cs:255]: should this line not check if the damage is positive and not just any value? You could theoretically have healing gas

**Review by EmoGarbage404 — APPROVED** [review]
merge at will lgtm

File: Content.Server/Body/Systems/RespiratorSystem.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Server/Body/Systems/RespiratorSystem.cs:255]: Its a poorly named method, ill rename it to `AnyPositive()` cause that's what its actually doing

**Pull request merged**